src/tests/fuzz/fuzz_decodebase64.c

```diff
@@ -0,0 +1,41 @@
+/**
+ * @file
+ * @author Shivani Bhardwaj <shivani@oisf.net>
+ * fuzz target for DecodeBase64
+ */
+
+#include "suricata-common.h"
+#include "suricata.h"
+#include "util-base64.h"
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size, size_t dest_size);
+
+static int initialized = 0;
+
+static void Base64FuzzTest(const uint8_t *src, size_t len, size_t dest_size)
+{
+    for (uint8_t mode = BASE64_MODE_RELAX; mode <= BASE64_MODE_RFC4648; mode++) {
+        uint8_t *dest = malloc(dest_size);
+        uint32_t consumed_bytes = 0;
+        uint32_t decoded_bytes = 0;
+
+        DecodeBase64(dest, dest_size, src, len, &consumed_bytes, &decoded_bytes, mode);
+    }
+}
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size, size_t dest_size)
```

Comment:
`LLVMFuzzerTestOneInput` must only have 2 arguments.

`dest_size` could be read from `data`...

Reply:
Interesting. I knowingly put the 3rd field incorrectly first to see if something compiles this in the CI and would fail and the Ubuntu fuzz jobs failed so, I thought I'll get an error in the CI if the fn signature is incorrect.

Q: I want `dest_size` to be randomized by fuzzer and not be in accordance with `data` as you had suggested earlier. How do I do that?

Reply:
@catenacyber ^^ Please take a look when you find time. Just making sure this question isn't lost and I can move forward with this work.

Reply:
Just read two bytes big or little endian in data ;-)

Reply:
Sorry, not following. If I have `d3d3LnN1cmljYXRhLmlv`, what do I achieve by reading two bytes big or little endian to get a randomized `dest_size`? 🤔

Reply:
See for example `f->sp = (uint16_t)((data[2] << 8) | data[3]);` in fuzz_applayerparserparse.c

Reply:
thanks! I did think of something like like but felt that's not what you meant. :)